var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -238,6 +238,8 @@ class Hdf5(CMakePackage):
     conflicts("+fortran", when="+shared@:1.8.15")
     # See https://github.com/spack/spack/issues/31085
     conflicts("+fortran+mpi", when="@1.8.22")
+    # See https://github.com/HDFGroup/hdf5/issues/2906#issue-1697749645
+    conflicts("+fortran", when="@1.13.3:^cmake@3.18:3.22", msg="cmake_minimum_required is not set correctly.")
```

Comment:
```suggestion
    conflicts("+fortran", when="@1.13.3:^cmake@:3.22", msg="cmake_minimum_required is not set correctly.")
```

Slightly simpler, should be equivalent, just less repetition.